### DISCUSSION TITLE
Update documentation for STI

### DISCUSSION THREAD

**Discussion by whitespacecode** [CodeDredd/pinia-orm#335] [General]
Maybe not such a big deal but currently the documentation for Single Table Inheritance makes it that the Models are in one file?
What if you rather have them split in separated model files.

The base entity should have `export { Class } `
And the derived entity should have an `import { Class } from './Class';`

**Comment by CodeDredd** [comment]
Thanks for the hint.

**Comment by n-xt** [comment]
Could you please explain the solution in more detail, I am getting a circular error here:

**Parent class:**

```
import Category from "./Category";

export default class Resource extends Model {
static entity = "resources";
...
static types() {
    return {
      CATEGORY: Category,
      ...
    };
}
```

**Sub class**

```
import Resource from "./Resource";

export default class Category extends Resource {
  static entity = "categories";
  static baseEntity = "resources";
   ...
}
```

**Reply by CodeDredd** [reply]
You need to use `export` in the classes also used in circular depenedicies and then import them as seperate module.

````ts
import Resource from "./Resource";

export default class Category extends Resource {
  static entity = "categories";
  static baseEntity = "resources";
   ...
}

export { Category }
````

````ts
import { Category } from "./Category";

export default class Resource extends Model {
static entity = "resources";
...
static types() {
    return {
      CATEGORY: Category,
      ...
    };
}
````
